odilo.py

```diff
@@ -147,18 +147,37 @@ def refresh_creds(self, credential):
         response = self.token_post(
             self.TOKEN_ENDPOINT,
             dict(grant_type="client_credentials"),
-            allowed_response_codes=[200]
+            allowed_response_codes=[200, 400]
         )
-        data = response.json()
+
+        # If you put in the wrong URL, this is where you'll run into
+        # problems, so it's useful to give a helpful error message if
+        # Odilo doesn't provide anything more specific.
+        generic_error = "%s may not be the right base URL. Response document was: %r" % (
+            self.library_api_base_url, response.content
+        )
+        generic_exception = BadResponseException(
+            self.TOKEN_ENDPOINT, generic_error
+        )
+
+        try:
+            data = response.json()
+        except ValueError:
+            raise generic_exception
```

Comment:
This handles a case I saw with a real user, who put in the URL to a library's Odilo OPAC instead of the library's Odilo API URL.